help/share-and-upload-files.md

```diff
@@ -1,21 +1,63 @@
 # Share and upload files
 
-Attach files to messages, including images, documents, sound, and video.
+Zulip supports attaching multiple files to messages, including images,
+documents, sound, and video. You can edit the names of the files others see
+after you upload them.
 
-First, [open the compose box](/help/open-the-compose-box). Then
+Zulip will automatically generate a **thumbnail** for each file when you send
+the message, if it can. Image thumbnails will be shown directly in the message,
+and you can click on a thumbnail to [view the full image](/help/view-and-browse-images).
 
-* **Drag and drop** files into the compose box.
-* **Copy and paste** files into the compose box.
-* Click the **paperclip** (<i class="fa fa-paperclip"></i>) icon at
-  the bottom of the compose box to find files on your computer.
+## Uploading a file
 
-Zulip will insert a link to the file, in Markdown format:
-`[link text](URL)`. You can modify the `link text` to whatever you want.
+{start_tabs}
 
-Zulip will automatically generate a **thumbnail** of the file when you send
-it, if it can. You can
-[preview the message](/help/preview-your-message-before-sending) before
-sending to see what the thumbnail will look like.
+{tab|via-markdown}
+
+{!start-composing.md!}
+
+1. Drag and drop or copy and paste files into the compose box to
+   upload and insert named links to the files:  
+   `[File-name](URL)`
+
+1. Modify the link text as desired.
+
```

Comment:
Let's make this "*(optional)* Modify the file name as desired.", to match the instruction above.

Reply:
We can do the same on the next tab.